### DISCUSSION TITLE
Update Opennept4une

### DISCUSSION THREAD

**Discussion by Justdigit** [OpenNeptune3D/OpenNept4une#119] [Q&A] [answered]
Can someone tell me what the best way is to upgrade the Neptune4??

I have updated some versions via Kiauh.sh but now I have an error message which I can't get rid of. So I will need to reflash the emmc. 

Do you update by kiauh?? Or via fluidd?? Or do you need to update the MCU? Via Openept4une gui??

Or is updating the MCU sufficient? 

Maybe something for the wiki.

**Comment by barrenechea** [comment]
Hi! eMMC is a prerequisite. Once you install the OpenNept4une image, you can do the updates through Fluidd/Mainsail. MCU flashing is required for using the latest Klipper versions. Otherwise, Klipper will fail with an error message.
Please check https://github.com/OpenNeptune3D/OpenNept4une/wiki#prerequisites and follow from there 😄

**Reply by Justdigit** [reply]
Yeah ok? .... So when there are update warnings in Fluidd/mainsail I can also choose to update the MCU? But not both....?? 
The MCU is only Klipper?? Or also things like Moonraker and so on?

**Reply by halfmanbear** [reply, marked as answer]
Your moonraker error is most likely this bug that appeared on mobileraker the other day.

This is the fix,
https://github.com/Clon1998/mobileraker_companion/issues/83

You can update modules via fluidd or kiauh, you can update the OS in either of these too. The MCU are updated via the OpenNept4une GUI as in the wiki.

**Reply by Justdigit** [reply]
Thanks!!! that wa the issue indeed....  I was confident i did something worng.
